tests/i2c_device/src/test.xc

```diff
@@ -56,6 +56,7 @@ void test_client(client interface control i[3], chanend c_user_task[3])
               tmr :> t;
               timeout = 0;
               par {
+#pragma warning disable unusual-code // Suppress slice interface warning (no array size passed)
```

Comment:
could you explain why this is needed?

Reply:
This is a warning that was introduced in some later versions of the tools, I guess it was 14.1.x but I am not sure. This warning should be addressed when the XC code of lib_device_control is updated/replaced.

If you don't suppress the warning, you get several warnings like the one below:
`warning: argument 1 of `control_process_spi_master_ends_transaction' slices interface preventing analysis of its parallel usage [-Wunusual-code]`